trunner/tools/gdb.py

```diff
@@ -94,30 +94,32 @@ class OpenocdError(ProcessError):
 class OpenocdGdbServer:
     """Handler for OpenocdGdbServer process"""
 
-    def __init__(self, interface: str, target: str):
+    def __init__(self, interface: str, target: str, extra_args: Optional[list[str]] = None):
```

Comment:
Then `if self.extra_args:` shouldn't be needed.

```suggestion
    def __init__(self, interface: str, target: str, extra_args: Optional[List[str]] = []):
```

Reply:
no.
https://www.codiga.io/blog/python-avoid-empty-array-parameter/

Reply:
That's right - I forgot about this issue, so we can leave this as it was.